DemoApp/middlewares/upload.js

```diff
@@ -0,0 +1,47 @@
+const util = require("util");
+const multer = require("multer");
+const maxSize = 2 * 1024 * 1024;
+const path = require('path');
+const uuid = require('uuid').v4;
+const Report=require('../models/Report');
+let storage = multer.diskStorage({
+  destination: (req, file, cb) => {
+    cb(null,"upload");
+    
+
+  },
+  filename: (req, file, cb) => {
+    console.log("dddddd"+file.originalname)
+    const ext=path.extname(file.originalname)
+    const id=uuid();
+    const filepath=`documents/${id}${ext}`; 
+    console.log(filepath)
+    Report.create({filepath})
+                .then(()=>{
+              cb(null,filepath)
+              })
+
+    console.log(file.originalname);
+    cb(null, file.originalname);
+  },
+  
+});
+
+
+const fileFilter = function(req, file, cb) {
+    // Accept doc|pdf only
+    if (!file.originalname.match(/\.(pdf|doc|docx)$/)) {
```

Comment:
Good job on extensions validations. Although, you might need to check an actual binary validator which check signature of if it's actually pdf or doc or any other file with wrong extensions.